File: nghiata/main4.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
    int calculate(string s) {
        int n = s.size(); 

        int ans = 0;
        int i = 0;
        while (i < n) {
            int j = i;
            while (j < n && s[i] == s[j]) {
                j++;
            }

            ans += floor((j - i) / 2);
            i = j;
        }

        return ans;
    }
public:
    vector<int> solution(vector<string>& arr) {
        vector<int> res = vector<int>(arr.size());
        for (int i = 0; i < arr.size(); i++) {
            string s = arr[i];
            res[i] = calculate(s);
        }

        return res;
    }
};

int main() {
    Solution sol;
    
    vector<string> testCases = {
        "aaaaaaa",
        "AABAAB",
        "ABAB",
        "AAAA",
        "BBAABB"
    };
    
    vector<int> results = sol.solution(testCases);
    
    cout << "Results for each test case:" << endl;
    for (int i = 0; i < results.size(); i++) {
        cout << "Test case \"" << testCases[i] << "\": " << results[i] << endl;
    }
    
    return 0;
}

File: nghiata/main4.go
package main

import (
	"fmt"
	"math"
)

func calculate(s string) int32 {
	n := len(s)
	var ans int32
	i := 0

	for i < n {
		j := i
		for j < n && s[i] == s[j] {
			j++
		}

		ans += int32(math.Floor(float64(j-i) / 2))
		i = j
	}

	return ans
}

func solution(arr []string) []int32 {
	res := make([]int32, len(arr))
	for i, s := range arr {
		res[i] = calculate(s)
	}
	return res
}

func main() {
	testCases := []string{
		"aaaaaaa",
		"AABAAB",
		"ABAB",
		"AAAA",
		"BBAABB",
	}

	results := solution(testCases)

	fmt.Println("Results for each test case:")
	for i, result := range results {
		fmt.Printf("Test case \"%s\": %d\n", testCases[i], result)
	}
}
